### ISSUE TITLE
This should be using `self.content`

### ISSUE THREAD

**Issue by BenGitsCode** [ga-wdi-boston/ruby-objects-self#25]
```rb
# frozen_string_literal: true

# for chaining methods
class MethodChainer
  attr_reader :content

  def initialize(str)
    @content = str
  end

  def reverse!
@content = @content.reverse
```

**Comment by payne-chris-r** [comment]
Do you mean like this?
```rb
# frozen_string_literal: true

# for chaining methods
class MethodChainer
  attr_reader :content

  def initialize(str)
    @content = str
  end

  def reverse!
    self.content = @content.reverse
```
???
Doesn't it need an attr_writer in order to have the `.content=()` method?

**Comment by BenGitsCode** [comment]
@payne-chris-r I think you're right, but this was poor issue-making on my part as I'm unclear what I meant and it looks like the perfect use case for a diff code block. I think you're right and I will look at it tonight when I'm on my computer and comment to at least my understand of the issue tonight as this should be merged by tomorrow morning.

**Comment by BenGitsCode** [comment]
@payne-chris-r have you looked at issue #20?

**Comment by BenGitsCode** [comment]
I think that was the catalysts for this issue.

**Comment by jrhorn424** [comment]
This is all correct. The point is comparing what happens when you assign to an attribute using self versus without. What would happen without?

Ruby would interpret content as a variable. <s>This is the only case I've come across where self is required.</s>

**Comment by gaand** [comment]
When self is required for instance methods?

**Comment by jrhorn424** [comment]
Yeah, I forgot about that and, of course, class method definitions. Comment updated.
